### PR TITLE
feat(cli): add SSL verification prompt with default to disabled

### PR DESCRIPTION
**Pull request by RaHehl**
### Description of change

- Add prompt for SSL certificate verification in CLI
- Change default from True to False (disabled by default)
- Rename flag from --no-verify to --verify-ssl for consistency
- Makes it clearer for users working with self-signed certificates

### Pull-Request Checklist

- [ ] Code is up-to-date with the `main` branch
- [ ] This pull request follows the [contributing guidelines](https://github.com/uilibs/uiprotect/blob/main/CONTRIBUTING.md).
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [ ] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/), such as "fix(api): prevent racing of requests".

> - If pre-commit.ci is failing, try `pre-commit run -a` for further information.
> - If CI / test is failing, try `poetry run pytest` for further information.

## Summary by CodeRabbit

* **New Features**
  * CLI flag renamed to --verify-ssl/--no-verify-ssl for clearer SSL verification control.

* **Bug Fixes**
  * Improved SSL error handling: detects verification failures, prompts to retry with verification disabled, and ensures proper cleanup on errors.

* **Documentation**
  * Updated SSL verification examples and guidance across Docker, CLI, and development docs.

* **Tests**
  * Added tests covering SSL error detection and related edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by bdraco** [comment]
The downside to this is we loose secure by default

File: docs/dev.md [hunk 1]
```diff
@@ -49,8 +49,8 @@ UFP_USERNAME=YOUR_USERNAME_HERE
 UFP_PASSWORD=YOUR_PASSWORD_HERE
 UFP_ADDRESS=YOUR_IP_ADDRESS
 UFP_PORT=443
-# change to false if you do not have a valid HTTPS Certificate for your instance
-UFP_SSL_VERIFY=True
+# set to true if you have a valid HTTPS certificate for your instance
+UFP_SSL_VERIFY=false
```
Comment (bdraco) [inline docs/dev.md:53]: Looks like UFP_SSL_VERIFY is no longer wired up?
